plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchCaretaker.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.swtbot;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.utils.SWTBotPreferences;
+import org.eclipse.swtbot.swt.finder.utils.SWTUtils;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+/** Take screenshot after test failure, and clean up the workbench. */
+public class SwtBotWorkbenchCaretaker extends TestWatcher {
+  private static final Logger logger = Logger.getLogger(SwtBotWorkbenchCaretaker.class.getName());
+
+  @Override
+  protected void starting(Description description) {
+    try {
+      SwtBotWorkbenchActions.closeWelcome(new SWTWorkbenchBot());
+    } catch (WidgetNotFoundException ex) {
```

Comment:
In cases like this, I think we usually name the exception variable "ignored" which will turn off some static analysis checks.